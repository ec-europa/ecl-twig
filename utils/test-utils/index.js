const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
