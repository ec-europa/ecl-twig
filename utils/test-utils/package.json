{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
