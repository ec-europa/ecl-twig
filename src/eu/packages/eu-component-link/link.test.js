import path from 'path';
import { renderTwigFile } from '@ecl-twig/test-utils';

describe('EU - Link', () => {
  const template = path.resolve(__dirname, './link.html.twig');
  const defaultIconPath = 'static/icons.svg';
  const defaultDataStructure = {
    link: {
      type: '',
      text: '',
      path: '/path',
    },
  };

  describe('Default', () => {
    test(`- link default renders correctly`, done => {
      expect.assertions(1);

      defaultDataStructure.link.type = 'default';
      defaultDataStructure.link.text = 'Default link';

      renderTwigFile(template, defaultDataStructure, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });

  describe('Standalone', () => {
    test(`- link standalone renders correctly`, done => {
      expect.assertions(1);

      defaultDataStructure.link.type = 'standalone';
      defaultDataStructure.link.text = 'Standalone link';

      renderTwigFile(template, defaultDataStructure, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });

  describe('With icon before', () => {
    test(`- link with icon renders correctly`, done => {
      expect.assertions(1);

      defaultDataStructure.link.type = 'standalone';
      defaultDataStructure.link.text = 'Standalone link with icon';
      defaultDataStructure.link.icon_position = 'before';
      defaultDataStructure.icon = {
        type: 'ui',
        name: 'external',
        size: 'fluid',
        path: defaultIconPath,
      };

      renderTwigFile(template, defaultDataStructure, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });

  describe('With icon after', () => {
    test(`- link with icon renders correctly`, done => {
      expect.assertions(1);

      defaultDataStructure.link.type = 'standalone';
      defaultDataStructure.link.text = 'Standalone link with icon';
      defaultDataStructure.link.icon_position = 'after';
      defaultDataStructure.icon = {
        type: 'ui',
        name: 'external',
        size: 'fluid',
        path: defaultIconPath,
      };

      renderTwigFile(template, defaultDataStructure, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });
});
