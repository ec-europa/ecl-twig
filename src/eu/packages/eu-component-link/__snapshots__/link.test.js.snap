// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EU - Link Default - link default renders correctly 1`] = `
"<a href=\\"/path\\" class=\\"ecl-link ecl-link--default\\"> </a>
"
`;

exports[`EU - Link Standalone - link standalone renders correctly 1`] = `
"<a href=\\"/path\\" class=\\"ecl-link ecl-link--standalone\\"> </a>
"
`;

exports[`EU - Link With icon after - link with icon renders correctly 1`] = `
"<a
  href=\\"/path\\"
  class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after\\"
>
  <span class=\\"ecl-link__label\\"></span>
  &nbsp;

  <svg
    class=\\"ecl-icon ecl-icon--fluid ecl-link__icon\\"
    focusable=\\"false\\"
    aria-hidden=\\"true\\"
  >
    <use xlink:href=\\"static/icons.svg#ui--external\\"></use>
  </svg>
</a>
"
`;

exports[`EU - Link With icon before - link with icon renders correctly 1`] = `
"<a
  href=\\"/path\\"
  class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
>
  <svg
    class=\\"ecl-icon ecl-icon--fluid ecl-link__icon\\"
    focusable=\\"false\\"
    aria-hidden=\\"true\\"
  >
    <use xlink:href=\\"static/icons.svg#ui--external\\"></use>
  </svg>
  &nbsp;
  <span class=\\"ecl-link__label\\"></span>
</a>
"
`;
